### ISSUE TITLE
RPC passwords for BitcoinNodes and LightningNodes should come from a Secret

### ISSUE THREAD

**Issue by davgordo** [kiln-fired/kiln-operator#27]
Currently the controllers set RPC passwords for BitcoinNodes and Lightning Nodes directly as environment variables. It would be better if they came from a Secret.

**Comment by thinkmassive** [comment]
[lndinit](https://github.com/lightninglabs/lndinit/) is a wallet initializer utility for `lnd` that also handles RPC secrets.

It looks like the controller uses an `lnd` image at present, so it the kubernetes example in that repo should be a good reference for changing to use the `lndinit` image. Otherwise the utility+script might be useful as a reference for adding similar functionality into the controller.

One cool feature I like about the example is how it uses a named pipe to load the wallet secret.

**Comment by davgordo** [comment]
> [lndinit](https://github.com/lightninglabs/lndinit/) is a wallet initializer utility for `lnd` that also handles RPC secrets.

Dang, I wish I found this earlier! All good, I may have written some duplicate implementation. I'll try to revisit. It was a good learning experience anyway. Cheers!

**Comment by davgordo** [comment]
@thinkmassive After review, thankfully, I think I mostly avoided overlap with `lndinit`. I think I would prefer to handle the RPC TLS stuff with [cert-manager](https://cert-manager.io/) as this is quite mature and well adopted, even allowing ACME or Vault integration with a root authority. That was one of the first things I tried out. I'm getting close to publishing a pretty complete demo though not quite there, but for just a rough glimpse at the cert-manager concept you take maybe get an idea from [here](https://github.com/davgordo/lnd-openshift-demo/tree/kiln). (The CRs are now out-of-date with the current kiln specification, so you won't be able to follow the readme without problems.)

However, I am really interested in `lndinit` for the wallet generator. If I'm understanding it correctly, it basically solves the last  piece of the puzzle for a major milestone I'm targeting. I've logged a separate issue (#31) to bootstrap the wallet.db. It's something I didn't have an implementation plan for yet, so that's very exciting. Thanks for the pointer.
